src/main/scala/com/sky/kafka/configurator/Main.scala

```diff
@@ -42,22 +43,26 @@ object Main extends LazyLogging {
     parse(args).flatMap { conf =>
       val configurator = TopicConfigurator.reader(conf)
       val result = configureTopicsFromFile(conf.file, configurator)
-
+        .recoverWith {
+          case NonFatal(throwable) =>
+            logger.error("Error whilst configuring topics", throwable)
+            Failure(throwable)
```

Comment:
Do we need to create a new object? Can't we just use the original failure?

Also do we want to log only the `NonFatal`? (not sure here... just asking!)